tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -38,8 +38,10 @@ object BuilderUtil {
     )(variable: TBuilderInstruction,
       set: TBuilderInstruction,
       expr: TBuilderInstruction): TBuilderInstruction = for {
+    usedBefore <- allUsed
     setEx <- set
-    usedInSet <- allUsed // variable may not appear as bound or free in set
+    usedInSetOrBefore <- allUsed // variable may not appear as bound or free in set
```

Comment:
Why does `allUsed` get bound to two different variables here?

Reply:
Because it's a getter of the internal state, which was changed in between by evaluating `set`.

Reply:
I think it makes sense to point this out in the Scaladoc of the shorthands, and to explain it in a short comment here.

Reply:
I can make it `getAllUsed`, np